Popup Windows - Portfolio Project

JuanDiegoGuerra opened the pull request:
- Created Popup Window for "My Recent Work" projects using **Javascript**.
  - Mobile version ✅
  - Desktop version ✅
> Sumary Preview 🕵🏼‍♂️:
 - Mobile Version 📱:

![mobile-version-popupwindow2](https://user-images.githubusercontent.com/45726631/231408092-a13ab786-150f-4a2b-b1e1-1ac0180cf11d.gif)
 - Desktop Version🖥:

![desktop-version-popupwindow](https://user-images.githubusercontent.com/45726631/231408132-bac92916-c68a-47c4-b273-9f3b8adf7bd1.gif)

JuanDiegoGuerra committed:
Update index.html

Changed projects titles/names

JuanDiegoGuerra committed:
Update index.html

Changed languages/skills names of the projects

JuanDiegoGuerra committed:
Add 1gif + 1img

Added 2 media files for Caesar Cipher Project preview

JuanDiegoGuerra committed:
Add 1gif + 1img

Added 2 media files for Chronometer project preview

JuanDiegoGuerra committed:
Create hello_microverse.JPG

Added 1 media files for Hello-Microverse project preview

JuanDiegoGuerra committed:
Add 1gif + 1img

Added 2 media files for Pokedex project preview

JuanDiegoGuerra committed:
Create setup_portfolio.JPG

Added 1 media file for Setup-Portfolio project preview

JuanDiegoGuerra committed:
Create try_boostrap.JPG

Added 1 media file for Try-Boostrap project preview

JuanDiegoGuerra committed:
Update index.html

Updated corresponding media files for project previews

JuanDiegoGuerra committed:
Create pokedex_preview2.gif

Added 1 media file for Pokedex project preview.

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed "Get my resume" button
Added Popup window container styles
Added "See project" container styles

JuanDiegoGuerra committed:
Update mobile_first.css

Added styles for "my works section"

JuanDiegoGuerra committed:
Update mobile_first.css

Added Popup window container styles

JuanDiegoGuerra committed:
Update mobile_first.css

Added stlyes to popup window elements

JuanDiegoGuerra committed:
Update mobile_first.css

Added styles for popup window container for DESKTOP SIZE

JuanDiegoGuerra committed:
Update mobile_first.css

Added styles for popup window elements for DESKTOP SIZE

JuanDiegoGuerra committed:
Update index.html

Linked font awesome stylesheet
Changed preview gif for Pokedex project
Changed Get my Resume class and id

JuanDiegoGuerra committed:
Update app.js

Create Object/Constructor for Project Cards
Added work 1 / project 1

JuanDiegoGuerra committed:
Update app.js

Added work 2 / project 2

JuanDiegoGuerra committed:
Update app.js

Added work 3 / project 3

JuanDiegoGuerra committed:
Update app.js

Added work 4 / project 4

JuanDiegoGuerra committed:
Update app.js

Added work 5 / project 5

JuanDiegoGuerra committed:
Update app.js

Added work 6 / project 6

CesarHerr committed:
Update app.js

Added pop up section to the .js file

CesarHerr committed:
Update app.js

Added popup elements from array object

CesarHerr committed:
Update app.js

Add close pop up function

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed popup window slider styles

JuanDiegoGuerra committed:
Update app.js

Fixed "works-tool section" class call

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed popup window for desktop size

JuanDiegoGuerra committed:
Update mobile_first.css

fixed styles for see live and see source buttons

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed elements of popup window desktop size

JuanDiegoGuerra committed:
Update app.js

Changed Languages names for projects 5 and 6
Added and id for project work name class

JuanDiegoGuerra committed:
Update mobile_first.css

Fixing styles

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed styles 2

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed background color to give that translucint effect

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed Linter erros

JuanDiegoGuerra committed:
Update app.js

Fixed Linter (Eslint JS) Erros

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed popup window close icon

joskalenda commented on js/app.js:
```diff
@@ -9,4 +9,127 @@ hamburger.addEventListener('click', () => {
\ No newline at end of file
+});
+
+const ProjectCardsDetails = [
+  {
+    id: 'work1',
```
- [ ] You have a good design touch but I think some improvements will be most welcome, stating on the mobile version, please take a look at these images

#### PROJECT CARD
<img width="732" alt="1" src="https://user-images.githubusercontent.com/77843050/231430646-aaec2458-569e-46ce-bdb3-c225c64244af.png">

- Am comparing your design to the figma one and we agree that there are some points to modify, right? please use the same style properties as provided in the design like the text colors, sizes, padding, margins, and font weight, make a copy of the image you see on the left as it represents the original template to follow. 
- Make sure you align your page in the middle of the page please, it seems like there is a big space on the right compared with the left one
- Yor logo has disappeared, please fix that.

JuanDiegoGuerra replied:
- I know there are lots of points to modify, but there are also things that I don't understand why I was asked to change before and now have to put it back as it was, and in this case would be the button style, I couldn't use the "white" font color because the color used on the design never passed the accessibility checks, and my reviewer from that activity told me to use a color with a different contrast, so I changed it and then all the accessibility checks were fine. And now I have to put the "white" font color back? so then It will look as the design again but won't be accessible? I don't know.

 - Another thing is that my project doesn't actually use a logo, I put it in when I started to built the project and I just haven't delete that section because I think I can use it later, when there won't be a review, like 2 that I had before, that tells me to remove it because my template doesn't have one. But now, that you pointed me that "my logo disappeared", a logo that doesn't exist in the template, I don't know if I should use it or not...

joskalenda commented on css/mobile_first.css:
```diff
@@ -614,6 +629,160 @@ footer h3:hover {
+
 /* ---  MEDIA QUERIES --- */
 
 @media screen and (max-width: 768px) {
```
- [ ] We are on this same media screen but the design seems to not respect it as well, please fix all the necessary modifications as you see the missfonctionment of the app in the record below
![5](https://user-images.githubusercontent.com/77843050/231432917-e1ef1511-2f96-4041-a411-1bc783c5600d.gif)

joskalenda commented on js/app.js:
```diff
@@ -9,4 +9,127 @@ hamburger.addEventListener('click', () => {
+// Pop Up section
+const PopupWindow = document.querySelectorAll('.external_link');
+const popUp = document.querySelector('.popup');
+
+PopupWindow.forEach((p, index) => {
```
- [ ] For the popup too, please take care of fixing the broken design, you need to stick as much as possible to the figma design, see this below
<img width="748" alt="2" src="https://user-images.githubusercontent.com/77843050/231432039-a919cb40-1d39-41d3-904d-3048840f817e.png">
- Text Color
- Alignments
- Fontweigt

On the desktop, it is more than this, see the below record
and update the style to match the given design
![6](https://user-images.githubusercontent.com/77843050/231432451-150b4ec1-bee6-4318-b962-d26dff1e2078.gif)

joskalenda commented on js/app.js:
```diff
@@ -9,4 +9,127 @@ hamburger.addEventListener('click', () => {
+          <li class="work-tool">${ProjectCardsDetails[index].languages[0]}</li>
+          <li class="work-tool">${ProjectCardsDetails[index].languages[1]}</li>
+          <li class="work-tool">${ProjectCardsDetails[index].languages[2]}</li>
+          <li class="work-tool">${ProjectCardsDetails[index].languages[3]}</li>
+        </ul>
+        <p class="notes">
+        ${ProjectCardsDetails[index].projectInfo}
+        </p>
```
- [ ] Personally, I think it is not a good idea of using a js array, it is efficient and easy to use the map method over it to get all its items then using indexes[0...], what if you have an array of 100 items, are you gonna do [0],[1],[n?]?   
use
``` 
project[index].map((item) => {
         <div> item.languages </div> 
#You can do the same for the rest of the items too
})
```

joskalenda reviewed (changes requested):
| STATUS: REQUIRED CHANGES ♻️  |
|--------------------------------|
| Hi Team, |
|Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there! |
| <img src="https://media4.giphy.com/media/CheWagmtEaw6TBd95F/giphy.gif"/> |

## Highlights

- [ ].. ✅ Green linters
- [ ].. ✅ Good commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

JuanDiegoGuerra committed:
Update app.js

Removed 1 unnecesary element from languages array.

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed styles

JuanDiegoGuerra committed:
Update animations_transitions.css

Added a transition for popup buttons

JuanDiegoGuerra committed:
Update mobile_first.css

Fixing font weight of Projects' names

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed parragraf and project title size

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed project tools styles

JuanDiegoGuerra committed:
Update app.js

Modified the project description to the one used in the template...

JuanDiegoGuerra committed:
Update mobile_first.css

fixed pop cointer size for mobile size

JuanDiegoGuerra committed:
Create Snapshoot_Portfolio.png

Project img for popup window

JuanDiegoGuerra committed:
Update mobile_first.css

modified popup window project name margin

JuanDiegoGuerra committed:
Create Snapshoot Portfolio.png

Added media file for project img for desktop size

JuanDiegoGuerra committed:
Update app.js

Fixed popup window img preview

JuanDiegoGuerra committed:
Update mobile_first.css

Added media query breakpoint

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed linters

JuanDiegoGuerra committed:
Update app.js

Fixed linters

JuanDiegoGuerra committed:
Update app.js

removing style that was declarated inside element

JuanDiegoGuerra committed:
Update app.js

added a class for links

JuanDiegoGuerra committed:
Update mobile_first.css

added styles for new link classes

JuanDiegoGuerra commented:
Requiered Changes made ✅

> Sumary Preview 🕵🏼‍♂️:

 ![final_preview](https://user-images.githubusercontent.com/45726631/231640935-0ee8bb26-a090-4924-9a69-e05643b2287a.gif)

 ![final_preview_desktop](https://user-images.githubusercontent.com/45726631/231641003-4b318a7f-54d0-4f94-941c-9f5e5d890ac0.gif)

![final_preview_desktop_responsiveness2](https://user-images.githubusercontent.com/45726631/231643210-837b2010-6e17-4d45-be63-76de53812d98.gif)

HAMMAS-SALEEM commented on css/animations_transitions.css:
```diff
@@ -46,6 +46,11 @@
   background: #008552;
 }
 
```
![image](https://user-images.githubusercontent.com/93039359/231651257-814bc150-ba74-44d7-b157-06ccbf5f3a75.png)

- [ ] It would be nice if you make your cards align in the center of the page and reduce some more space from the right side :+1: as suggested by the previous reviewer to make your project follow the Figma template to make your project professional :blush:
- [ ] Kindly make your title **(Pokedex Personal Work)** color as you can see in the Figma template as suggested by the previous reviewer :+1:

HAMMAS-SALEEM replied:
![image](https://user-images.githubusercontent.com/93039359/231652438-e7130ee9-7f2f-4e08-85e9-2c5e02c057f7.png)

- [ ] It would be nice if you make your popup description color follow the Figma template as suggested by the previous reviewer :+1:

HAMMAS-SALEEM commented on css/animations_transitions.css:
```diff
@@ -46,6 +46,11 @@
+}
+
 /* ----- TOP MENU (Portfolio/About/Contact) ----- */
 @media screen and (min-width: 768px) {
   .list-item {
```
![image](https://user-images.githubusercontent.com/93039359/231652004-fb546d6a-c3f3-4d1f-bcf1-0efb6bb7cc0b.png)

- [ ] As suggested by the previous reviewer that your application has some horizontal scroll :fearful: Kindly remove that to make your project look professional and follow the Figma template :+1:

HAMMAS-SALEEM commented on js/app.js:
```diff
@@ -16,7 +16,8 @@ const ProjectCardsDetails = [
-      'Hello-Microverse-Project is mean to be an example of how to set up a simple project using git & github best practices.',
+        'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s.Lorem Ipsum has been the industrys standard dummy text ever.',
+      //'Hello-Microverse-Project is mean to be an example of how to set up a simple project using git & github best practices.',
     ProjectIMG: [
```
- [ ] It would be nice if you remove all the commented pieces of code from your app.js file to make the app.js file clean :+1: because you have added a lot of commented piece of code while making the requested changes :blush:

HAMMAS-SALEEM commented on js/app.js:
```diff
@@ -81,7 +86,8 @@ const ProjectCardsDetails = [
     id: 'work6',
```
![image](https://user-images.githubusercontent.com/93039359/231653944-19089fa7-f62c-4218-a26d-eeb68066bba1.png)

**[OPTIONAL]**

- [ ] It would be nice if you remove all the section descriptions, and unnecessary data from your readme file to make your readme file descriptive and clean :+1:

HAMMAS-SALEEM reviewed (changes requested):
# Required Changes :recycle:
![almost-there](https://user-images.githubusercontent.com/65860678/170491251-b1add8ba-b18f-4648-a893-20e510d957fd.gif)

## Hey @JuanDiegoGuerra :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requested changes, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers, and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

JuanDiegoGuerra committed:
Update app.js

Removed commented pieces of code.

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed popup description and title colors.

JuanDiegoGuerra committed:
Update README.md

Updated README content such as Promgramming tools, Key features, Future Features...

JuanDiegoGuerra committed:
Update index.html

Added a wrapper to fix overflowing elements across the html

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed unwated whitespace in the right side of the page caused by overflowing elements.

JuanDiegoGuerra committed:
Update mobile_first.css

Fixed Linters

JuanDiegoGuerra commented:
###  Required Changes Made 🛠: 

1.  Remove section description from Readme ✅
2.  Remove all the commented pieces of code from JS file✅
3.  Fixed project description color ✅
4.  Fixed project title color ✅
5.  Remove horizontal scroll   _( ᴵᵐ ʰᵃᵖᵖʸ ᴵ ᶜᵒᵘˡᵈ ᶠᶦˣ ᵗʰᶦˢ 🤩)_ ✅
6.  Align project cards and reduce space from right side ✅

![todolist1](https://user-images.githubusercontent.com/45726631/231703283-083f07c2-f037-4fa4-a834-be7d3bd26e06.gif)

>NOTE: Thank you guys for the feedback, I'm very motivated now!

AdnanOlarmmi reviewed (approved):
Hi @JuanDiegoGuerra,

This is Adnan Is-haq, your beloved code reviewer, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![Approved office man](https://user-images.githubusercontent.com/41163571/219113922-5007d6bb-ff59-4916-b6d3-e5aad3cb4051.gif)

## Highlights
- Nice design.
- Fixed required changes.

## Optional suggestions
_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
